packages/function-tree-firebase-admin/src/QueueHandler.js

```diff
@@ -5,14 +5,15 @@ function authenticate ({firebase, props, path}) {
     .then((decodedToken) => ({uid: decodedToken.uid}))
 }
 
-function createRunTask (task, cb) {
+function noop () {}
+
+function createRunTask (task, shouldAuthenticate, cb) {
   return (data, progress, resolve, reject) => {
     const _execution = data._execution
 
     delete data._execution
-
     cb(task.specId, [
-      authenticate,
+      shouldAuthenticate ? authenticate : noop,
```

Comment:
Instead of having an empty function popping up in the debugger, this should rather be something like:

```js
[].concat(
  shouldAuthenticate ? authenticate : []
).concat(task.tree)
```